frontend/apps/reader/modules/readerview.lua

```diff
@@ -792,18 +793,27 @@ function ReaderView:onReadSettings(config)
     self.highlight.saved = config:readSetting("highlight", {})
     -- Highlight formats in crengine and mupdf are incompatible.
     -- Backup highlights when the document is opened with incompatible engine.
-    if #self.highlight.saved > 0 then
-        if self.ui.rolling and type(self.highlight.saved[1][1].pos0) == "table" then
+    if util.tableSize(self.highlight.saved) > 0 then
+        local highlight_type = type(select(2, next(self.highlight.saved))[1].pos0)
```

Comment:
Just `if next(self.highlight.saved) then` is cheaper than `util.tableSize(...)`.

Or for readability (because I had to lookup the doc for `next()` to understand why you were using `select(2, ...)`)
```lua
local some_page_num, some_page_highlights = next(self.highlight.saved)
if some_page_highlights then
    local highlight_type = type(some_page_highlights[1].pos0)
```

Are we sure that when we delete the last highlight of page 5, `self.highlight.saved[5]` disappears, and that it doesn't stay as `self.highlight.saved[5] = {}`

Reply:
> Are we sure that when we delete the last highlight of page 5, `self.highlight.saved[5]` disappears, and that it doesn't stay as `self.highlight.saved[5] = {}`

https://github.com/koreader/koreader/blob/a2d95f5a3f7d6132452fe38b31e0655524132d52/frontend/apps/reader/modules/readerhighlight.lua#L1623-L1627